src/components/avatar-rectangle/avatar-rectangle.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState, type FC, type HTMLAttributes } from 'react'
+import {
+  ElAvatarRectangle,
+  ElAvatarRectBottomPlaceholder,
+  ElAvatarRectBottomSmallPlaceholder,
+  ElAvatarRectCommercialPlaceholder,
+  ElAvatarRectCommercialSmallPlaceholder,
+  ElAvatarRectResidentialPlaceholder,
+  ElAvatarRectResidentialSmallPlaceholder,
+} from './styles'
+
+export interface AvatarRectangle extends HTMLAttributes<HTMLDivElement> {
+  variant: 'residential' | 'commercial'
+  size: 'medium' | 'small'
+  src?: string
+  alt?: string
+}
+
+export const AvatarRectangle: FC<AvatarRectangle> = ({
+  size = 'medium',
+  variant = 'residential',
+  src,
+  alt,
+  ...props
+}) => {
+  const [isError, setIsError] = useState(false)
+
+  const handleError = () => {
+    setIsError(true)
+  }
+
+  useEffect(() => {
+    if (src) {
+      setIsError(false)
+    }
+  }, [src])
+
+  const hasImageError = !src || isError
+
+  const PlaceholderComponent =
+    variant == 'residential'
+      ? size === 'medium'
+        ? ElAvatarRectResidentialPlaceholder
+        : ElAvatarRectResidentialSmallPlaceholder
+      : size === 'medium'
+        ? ElAvatarRectCommercialPlaceholder
+        : ElAvatarRectCommercialSmallPlaceholder
+
+  const CommercialBottomPlaceholder =
+    size === 'medium' ? ElAvatarRectBottomPlaceholder : ElAvatarRectBottomSmallPlaceholder
```

Comment:
actually commercial image still need the bottom placeholder

![image](https://github.com/user-attachments/assets/e8a395fa-18e3-4875-9b65-5f27d83011ed)

Reply:
Oohhh, right... 😞 I had completely forgotten that was still displayed when there was an actual image in use.

In hindsight, I think we could have always rendered the placeholder and just shown the image absolutely positioned over the top of it. All good though, what you've done is fine, so let's move on.

**todo:** it may be worth a comment in the code where the bottom placeholder is rendered just to make sure other readers in future don't get confused about why it's needed.